src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/DefaultFileUriProviderTest.cs

```diff
@@ -12,11 +13,72 @@ public class DefaultFileUriProviderTest
     {
         public DefaultFileUriProviderTest()
         {
-            TextBuffer = Mock.Of<ITextBuffer>(buffer => buffer.Properties == new PropertyCollection());
+            TextBuffer = new TestTextBuffer(StringTextSnapshot.Empty);
         }
 
         private ITextBuffer TextBuffer { get; }
 
+        [Fact]
+        public void AddOrUpdate_Adds()
+        {
+            // Arrange
+            var expectedUri = new Uri("C:/path/to/file.razor");
+            var uriProvider = new DefaultFileUriProvider(Mock.Of<ITextDocumentFactoryService>());
+
+            // Act
+            uriProvider.AddOrUpdate(TextBuffer, expectedUri);
+
+            // Assert
+            Assert.True(uriProvider.TryGet(TextBuffer, out var uri));
+            Assert.Same(expectedUri, uri);
+        }
+
+        [Fact]
+        public void AddOrUpdate_Updates()
+        {
+            // Arrange
+            var expectedUri = new Uri("C:/path/to/file.razor");
+            var uriProvider = new DefaultFileUriProvider(Mock.Of<ITextDocumentFactoryService>());
+            uriProvider.AddOrUpdate(TextBuffer, new Uri("C:/original/uri.razor"));
+
+            // Act
+            uriProvider.AddOrUpdate(TextBuffer, expectedUri);
+
+            // Assert
+            Assert.True(uriProvider.TryGet(TextBuffer, out var uri));
+            Assert.Same(expectedUri, uri);
+        }
+
+        [Fact]
+        public void TryGet_Exists_ReturnsTrue()
```

Comment:
I'm not really complaining since it implies an intentionality, but isn't this the same test as `AddOrUpdate_Adds`?

Reply:
Yup pretty much. Just testing it for exhaustiveness since it was a relatively easy test to add and will cover us if we ever change `AddOrUpdate`